test/integration/auth/auth.prose.test.ts

```diff
@@ -0,0 +1,373 @@
+        });
+      }
+
+      /**
+       * Step 2
+       * For a test user supporting both SCRAM-SHA-1 and SCRAM-SHA-256, drivers should verify
+       * that negotation selects SCRAM-SHA-256. This may require monkey patching, manual log
+       * analysis, etc.
+       * todo(NODE-5629): Test passes locally but will fail on CI runs.
+       */
+      it.skip('selects SCRAM-SHA-256 for a user that supports both auth mechanisms', {
+        metadata: { requires: { mongodb: '>=3.7.3' } },
+        test: async function () {
+          const options = {
+            auth: {
+              username: userMap.both.username,
+              password: userMap.both.password
+            },
+            authSource: this.configuration.db
+          };
+
+          const spy = sinon.spy(ScramSHA256.prototype, 'auth');
+
+          const client = this.configuration.newClient({}, options);
+          const stats = await client.db('test').stats();
+          expect(stats).to.exist;
+          expect(spy.called).to.equal(true);
+          sinon.restore();
+          await client.close();
+        }
+      });
+
+      /**
+       * Step 3
+       * For test users that support only one mechanism, verify that explictly specifying
+       * the other mechanism fails.
+       */
+      it('fails to connect if incorrect auth mechanism is explicitly specified', {
+        metadata: { requires: { mongodb: '>=3.7.3' } },
+        test: async function () {
+          const options = {
+            auth: {
+              username: userMap.sha256.username,
+              password: userMap.sha256.password
+            },
+            authSource: 'admin',
+            authMechanism: 'SCRAM-SHA-1'
+          };
+
+          const client = this.configuration.newClient({}, options);
+          const error = await client
+            .db('test')
+            .stats()
+            .catch(e => e);
+          expect(error.message).to.match(/Authentication failed|SCRAM/);
+          await client.close();
+        }
+      });
+
+      /*
+       * Step 3
+       * For a non-existent username, verify that not specifying a mechanism when
+       * connecting fails with the same error type that would occur with a correct
+       * username but incorrect password or mechanism.  (Because negotiation with a
+       * non-existent user name at one point during server development caused a
+       * handshake error, we want to verify this is seen by users as similar to other
+       * authentication errors, not as a network or database command error on the ``hello``
+       * or legacy hello commands themselves.)
+       */
+      it('fails for a nonexistent username with same error type as bad password', {
+        metadata: { requires: { mongodb: '>=3.7.3' } },
+        test: async function () {
+          const noUsernameOptions = {
+            auth: {
+              username: 'roth',
+              password: 'pencil'
+            },
+            authSource: 'admin'
+          };
```

Comment:
Am I reading this test incorrectly?  Looks like this still specifies `username`

Reply:
Removed username and updated test assertions with a comment since the Node driver fails validation when instantiating the client with no username present.